datus/utils/traceable_utils.py

```
@@ -19,39 +17,74 @@
+            def __init__(self, *args, **kwargs):
+                super().__init__(*args, **kwargs)
+                self._last_trace_url: str | None = None
+
+            def on_trace_end(self, trace) -> None:
+                # Capture trace URL from RunTree before super() pops it
+                run = self._runs.get(trace.trace_id)
+                if run:
+                    try:
+                        self._last_trace_url = run.get_url()
+                        logger.info(f"LangSmith Trace: {self._last_trace_url}")
+                    except Exception as e:
```

Ruff (0.14.14)

[warning] Do not catch blind exception: `Exception`

(BLE001)